### PR TITLE
Fix tag calculation if weekday is before imageDay

### PR DESCRIPTION
**Pull request by bastjan**
This PR fixes the day calculation if the weekday is before the imageDay and the window has not yet been met

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [ ] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [ ] Link this PR to related issues.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -88,14 +89,21 @@ func (t *tagHandler) getWindow(w http.ResponseWriter, r *http.Request) {
...
-		w.WriteHeader(http.StatusMethodNotAllowed)
+		w.WriteHeader(http.StatusUnprocessableEntity)
 		errMessage := fmt.Sprintf("error parsing hour")
 		_, err := io.WriteString(w, errMessage)
 		fmt.Println(err, errMessage)
 		return
 	}
 
-	tag := t.getTag(day, hour, time.Now())
+	tag, err := t.getTag(day, hour, time.Now())
```
Comment (bastjan) [inline main.go:99]: Sorry for the unrelated changes. The calculations give an undefined result with invalid params and should thus return an error.

File: main.go [hunk 2]
```diff
@@ -134,27 +140,27 @@ func (t *tagHandler) getTag(day int, hour int, currentTime time.Time) string {
...
 
-//getPreviousTag returns last week's tag according to the imageDay
+// getPreviousTag returns last week's tag according to the imageDay
 func (t *tagHandler) getPreviousTag(currentTime time.Time) string {
-	date := currentTime.AddDate(0, 0, -7)
-
-	return t.getImageDate(date)
+	if int(currentTime.Weekday()) < imageDay {
+		return t.getCurrentTag(currentTime)
+	}
```
Comment (simu) [inline main.go:157]: Might make sense to add a comment describing why this bit is needed. I would make a suggestion but I couldn't figure out a concise way to describe the reason.

**Review by simu — APPROVED** [review]

File: main.go (hunk 2 above)
Comment (bastjan) [reply main.go:157]: I'm also struggling to think of a concise explanation 😅
